std/experimental/allocator/building_blocks/null_allocator.d

```diff
@@ -79,7 +80,7 @@ struct NullAllocator
 {
     assert(NullAllocator.instance.alignedAllocate(100, 0) is null);
     assert(NullAllocator.instance.allocateAll() is null);
-    auto b = NullAllocator.instance.allocate(100);
+    auto b = (() nothrow @safe @nogc => NullAllocator.instance.allocate(100))();
```

Comment:
`pure`?

Reply:
This is not `pure` because it is using `instance` which is global data. I believe `pure` should accept  `stateless` global data, but it doesn't (for the time being at least)